tests/functional/firefox/test_new.py

```diff
@@ -0,0 +1,27 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from pages.firefox.new import FirefoxNewPage
+
+
+# We don't currently show the download button for up-to-date Firefox users.
+# TODO explore how could we run this test using a known out-of-date Firefox?
+@pytest.mark.skip_if_capabilities_include({'browserName': 'firefox'})
```

Comment:
Seeing this again, I wonder if it would be better to have marks like `skip_if_firefox`, which can then call common code with appropriate capabilities values to determine if we should skip. This would still give the same flexibility, at the cost of writing code to deal with these marks, but would make the tests more readable. We could then have something like `skip_if_latest_firefox` to assist with the TODO. These marks could then be combined on a single test, such as:

```python
@pytest.mark.skip_if_not_firefox
@pytest.mark.skip_if_latest_firefox
def test_click_download_button(base_url, selenium):
    page = FirefoxNewPage(base_url, selenium, params='').open()
    assert page.is_download_button_displayed
    page.click_download_button()
    assert page.is_thank_you_message_displayed
```

Of course, there are many ways we could slice this. Another option is to provide marks like `skip_if_browser_name` and take the browser name as an argument. That's more readable than the current code, but has its own costs. Ultimately, I'm happy to land something that works, and iterate on it.

Another thing I'm a little uneasy about is that we're currently using the actual capabilities, which is the safest way to know we're skipping for the right reason, but does require us to start a Selenium session (and browser). If we trusted our requested capabilities, we could even skip starting the unnecessary browser instance.

If he's around, I'd love to hear @bobsilverberg's thoughts on this too.